new-log-viewer/src/contexts/StateContextProvider.tsx

```diff
@@ -170,6 +174,45 @@ const loadPageByCursor = (
     });
 };
 
+/**
+ * If the new log event number is within the page boundaries, update the url with the nearest log
+ * event and return true. If the new log event number is outside page boundaries, do not update and
+ * return false.
+ *
+ * @param logEventNum
+ * @param logEventNumsOnPage
+ * @return Whether the log event number is within page boundaries.
+ */
+const updateUrlIfEventOnPage = (
+    logEventNum: number,
+    logEventNumsOnPage: number[]
+): boolean => {
+    if (false === isWithinBounds(logEventNumsOnPage, logEventNum)) {
+        return false;
+    }
+
+    // Given `isWithinBounds` returns `true`, then `logEventNumsOnPage` must bound `logEventNum` and
+    // `logEventNumsOnPage` cannot be empty. These assertions justify type casts at end of
+    // function.
+
+    const nearestIdx = findNearestLessThanOrEqualElement(
+        logEventNumsOnPage,
+        logEventNum
+    );
+
+    // First explicit cast since typescript thinks `nearestIdx` can be null, but
+    // it can't as `logEventNum` must be bounded by in `logEventNumsOnPage`.
+    // Second explicit cast since typescript thinks `logEventNumsOnPage` can be empty, but
+    // that's not possible in this branch.
```

Comment:
```suggestion
    const nearestIdx = findNearestLessThanOrEqualElement(
        logEventNumsOnPage,
        logEventNum
    );

    // Since `isWithinBounds` returned `true`, then:
    // - `logEventNumsOnPage` must bound `logEventNum`.
    // - `logEventNumsOnPage` cannot be empty.
    // - `nearestIdx` cannot be `null`.
    //
    // Therefore, we can safely cast:
    // - `nearestIdx` from `Nullable<number>` to `number`.
    // - `logEventNumsOnPage[nearestIdx]` from `number | undefined` to `number`.
```